app-modules/engagement/src/Policies/EngagementPolicy.php

```diff
@@ -38,10 +38,21 @@
 
 use App\Models\Authenticatable;
 use Illuminate\Auth\Access\Response;
+use AdvisingApp\Prospect\Models\Prospect;
 use AdvisingApp\Engagement\Models\Engagement;
+use AdvisingApp\StudentDataModel\Models\Student;
 
 class EngagementPolicy
 {
+    public function before(Authenticatable $authenticatable): ?Response
+    {
+        if (! $authenticatable->hasAnyLicense([Student::getLicenseType(), Prospect::getLicenseType()])) {
+            return Response::deny('You do not have permission to view students or prospects.');
```

Comment:
Same thought to make this generate as above

Reply:
done